endToEndTests/test/queries/stringSearch_basic_regex.json

```diff
@@ -0,0 +1,21 @@
+{
+  "testCaseName": "StringSearch with a basic regex",
+  "query": {
+    "action": {
+      "type": "Aggregated"
+    },
+    "filterExpression": {
+      "type": "Not",
+      "child": {
+        "type": "StringSearch",
+        "column": "gisaid_epi_isl",
+        "searchExpression": ".*EPI.*"
```

Comment:
Some statement that matches are rare, but one does exist in Java: "Yes, [Java's] matches() always acts as if the regex were anchored at both ends. To get the traditional behavior, which is to match any substring of the target, you have to use find() (as others have already pointed out). Very few regex tools offer anything equivalent to Java's matches() methods, so your confusion is justified. The only other one I can think of offhand is the [XML Schema](http://www.regular-expressions.info/xml.html) flavor."

https://stackoverflow.com/questions/1894624/is-regex-in-java-anchored-by-default-with-both-a-and-character

Reply:
Other thoughts:

* I think just because SQL `LIKE` implicitly anchors (and doesn't have anchoring syntax) is not enough of a reason to deviate from the usual approach with regular expressions. You could deviate, but I think you need another reason to justify it, and we should state the implicit anchoring in the docs with emphasis, and that reason.

* If it is to be made configurable, I suggest to do it in the following way:

    * In the frontend, have a checkbox *right next* to the entry field for the regex; have it *checked* for implicit anchoring (it *adds* something to normal regex behaviour; it could be checked by default if deemed right, though).
    * Instead of adding two modes to the backend, if the checkbox is checked, have the frontend surround the regex with `^` and `$` before sending the query to the backend. (Because it simplifies the backend and protocol, and the "mode" is always directly kept with the regex, meaning if e.g. the regex is printed to a log file, that "mode" is automatically part of it.) Or, being more explicit to the user, have an "anchor" button that inserts `^` and `$` visibly into the regex input field.

Reply:
After some considerations, also looking at similar implementations, we decided for unanchored regexes by default.